CTAP2 canonical form CBOR encoding parsing

reynir opened the pull request:
This PR adds a submodule `CBOR.Ctap2_canonical` with a similar signature as `CBOR.Simple` with its own `type t` without `` `Tag`` and a function `to_simple`, but no `to_diagnostic`. What is missing is `encode`.

I took some (mostly) stylistic choices that differ from the existing code - I can adapt it to fit the existing style.

Fixes #18 (parsing only)

The tests cases test both `CBOR.Simple` and `CBOR.Ctap2_canonical` simultaneously. They probably shouldn't.

reynir committed:
Add CTAP2 canonical form CBOR encoding parsing

ygrek commented on test/appendix_a.json:
```diff
@@ -69,6 +69,7 @@
     "cbor": "wkkBAAAAAAAAAAA=",
     "hex": "c249010000000000000000",
     "roundtrip": true,
+    "noncanonical": true,
```
iirc this json was copied straight from the rfc, now this change introduces a (small) diff

ygrek commented on test/appendix_a.json:
```diff
@@ -628,9 +646,75 @@
       "Amt": -2
     }
+  },
+  {
```
lets rather have new testcases in a separate file

ygrek reviewed (changes requested):
Thanks and sorry for the late review.
My only concern is changes to the test json, I think we can use test case numbers for ctap2 tests (same as it is done for ignore) from appendix_a and add ctap2-only test cases in separate json

reynir committed:
Revert appendix_a.json, new tests in ctap2.json

reynir committed:
Rework test harness

reynir commented:
I reverted appendix_a.json and moved the new tests to ctap2.json. I made some changes to the test harness that I'm not too happy about -- it compares the file name of the test json file in order to decide what tests are noncanonical or ignored in appendix_a.json.

Perhaps I should write two separate test programs rather. I will not do it today..

ygrek commented:
lgtm, thanks!

ygrek commented:
one thing I missed, I think module should be named CTAP2_canonical, do you agree?

ygrek commented:
https://github.com/ygrek/ocaml-cbor/commit/480a743fd5a03237bc488a7aec72e102894edee9
